$NetBSD: patch-configure.ac,v 1.3 2017/12/13 21:15:12 maya Exp $

Assume all other operating systems are nix-like
(SmartOS, dragonflybsd)

Improve portability of test (==)

Don't let package add its optimization flags

--- configure.ac.orig	2010-03-07 12:22:36.000000000 +0000
+++ configure.ac
@@ -319,7 +319,7 @@ case "$host_os" in
         ;;
 
     *)
-        hc_cv_is_nix=no
+        hc_cv_is_nix=yes
         hc_cv_is_windows=no
         hc_cv_is_mingw32=no
         hc_cv_is_apple=no
@@ -546,7 +546,7 @@ AC_CHECK_DECLS( SIOCADDRT,            [h
 AC_CHECK_DECLS( SIOCDELRT,            [hc_cv_have_siocdelrt=yes],           [hc_cv_have_siocdelrt=no],           [#include <linux/sockios.h>] )
 AC_CHECK_DECLS( SIOCDIFADDR,          [hc_cv_have_siocdifaddr=yes],         [hc_cv_have_siocdifaddr=no],         [#include <linux/sockios.h>] )
 
-if test "$hc_cv_have_sys_mtio_h" == "yes"; then
+if test "$hc_cv_have_sys_mtio_h" = "yes"; then
     AC_CHECK_DECLS( MTEWARN,          [hc_cv_have_mtewarn=yes],             [hc_cv_have_mtewarn=no],             [#include <sys/mtio.h>] )
 else
     hc_cv_have_mtewarn=no
@@ -1742,7 +1742,7 @@ if test "$hc_cv_auto_optimize" = "yes"; 
         xscale-yes|arm*-yes)
 
             hc_cv_is_intel_x86_arch=no
-            hc_cv_optimization_flags="$hc_cv_optimization_flags -mcpu=$host_cpu -mtune=$host_cpu -frename-registers"
+            hc_cv_optimization_flags="$hc_cv_optimization_flags -frename-registers"
             ;;
     esac
     if test "$hc_cv_is_intel_x86_arch" = "yes"; then
